### DISCUSSION TITLE
Why only videos without timezone metadata falls back to UTC?

### DISCUSSION THREAD

**Discussion by namu-lee** [immich-app/immich#7701] [Q&A]
Hello.

Only videos without timezone metadata are displayed in the UTC timezone.
Photos with no timezone metadata are correctly displayed in the fallback timezone(TZ).
I had set TZ variable in .env file as my local timezone.
Is it an intended behavior, or a bug?

Thanks.

**Comment by mmomjian** [comment]
I have noticed a similar issue but I haven't been able to pin down exactly what causes it. Either way, videos are much more likely to end up with the wrong metadata on my instance. Would be glad to provide some logs if it can help.

**Reply by namu-lee** [reply]
### Logs generated when a video was uploaded
immich_microservices     | [Nest] 7  - 03/08/2024, 12:35:52 PM VERBOSE [MetadataService] Exif Tags
immich_microservices     | [Nest] 7  - 03/08/2024, 12:35:52 PM VERBOSE [MetadataService] Object:
immich_microservices     | {
immich_microservices     |   "SourceFile": "/usr/src/app/upload/upload/76a54184-6569-4d47-8c94-1bb24232f6da/6b/9d/6b9d5c77-d8c1-401c-9622-512183b06230.mp4",
immich_microservices     |   "errors": [],
immich_microservices     |   "tz": "UTC",
immich_microservices     |   "tzSource": "defaultVideosToUTC",
immich_microservices     |   "Duration": 5.312,
immich_microservices     |   "PreviewDuration": 0,
immich_microservices     |   "SelectionDuration": 0,
immich_microservices     |   "TrackDuration": 5.28,
immich_microservices     |   "MediaDuration": 5.312,
immich_microservices     |   "ExifToolVersion": 12.76,
immich_microservices     |   "FileName": "6b9d5c77-d8c1-401c-9622-512183b06230.mp4",
immich_microservices     |   "Directory": "/usr/src/app/upload/upload/76a54184-6569-4d47-8c94-1bb24232f6da/6b/9d",
immich_microservices     |   "FileSize": "1056 kB",
immich_microservices     |   "FileModifyDate": {
immich_microservices     |     "_ctor": "ExifDateTime",
immich_microservices     |     "year": 2024,
immich_microservices     |     "month": 3,
immich_microservices     |     "day": 7,
immich_microservices     |     "hour": 19,
immich_microservices     |     "minute": 53,
immich_microservices     |     "second": 10,
immich_microservices     |     "tzoffsetMinutes": 540,
immich_microservices     |     "rawValue": "2024:03:07 19:53:10+09:00",
immich_microservices     |     "zoneName": "UTC+9",
immich_microservices     |     "inferredZone": false
immich_microservices     |   },
immich_microservices     |   "FileAccessDate": {
immich_microservices     |     "_ctor": "ExifDateTime",
immich_microservices     |     "year": 2024,
immich_microservices     |     "month": 3,
immich_microservices     |     "day": 8,
immich_microservices     |     "hour": 12,
immich_microservices     |     "minute": 35,
immich_microservices     |     "second": 51,
immich_microservices     |     "tzoffsetMinutes": 540,
immich_microservices     |     "rawValue": "2024:03:08 12:35:51+09:00",
immich_microservices     |     "zoneName": "UTC+9",
immich_microservices     |     "inferredZone": false
immich_microservices     |   },
immich_microservices     |   "FileInodeChangeDate": {
immich_microservices     |     "_ctor": "ExifDateTime",
immich_microservices     |     "year": 2024,
immich_microservices     |     "month": 3,
immich_microservices     |     "day": 8,
immich_microservices     |     "hour": 12,
immich_microservices     |     "minute": 35,
immich_microservices     |     "second": 51,
immich_microservices     |     "tzoffsetMinutes": 540,
immich_microservices     |     "rawValue": "2024:03:08 12:35:51+09:00",
immich_microservices     |     "zoneName": "UTC+9",
immich_microservices     |     "inferredZone": false
immich_microservices     |   },
immich_microservices     |   "FilePermissions": "-rw-r--r--",
immich_microservices     |   "FileType": "MP4",
immich_microservices     |   "FileTypeExtension": "mp4",
immich_microservices     |   "MIMEType": "video/mp4",
immich_microservices     |   "MajorBrand": "MP4 Base Media v1 [IS0 14496-12:2003]",
immich_microservices     |   "MinorVersion": "0.2.0",
immich_microservices     |   "CompatibleBrands": [
immich_microservices     |     "isom",
immich_microservices     |     "iso2",
immich_microservices     |     "avc1",
immich_microservices     |     "mp41"
immich_microservices     |   ],
immich_microservices     |   "MediaDataSize": 0,
immich_microservices     |   "MediaDataOffset": 1051515,
immich_microservices     |   "MovieHeaderVersion": 0,
immich_microservices     |   "CreateDate": "0000:00:00 00:00:00",
immich_microservices     |   "ModifyDate": {
immich_microservices     |     "_ctor": "ExifDateTime",
immich_microservices     |     "year": 2014,
immich_microservices     |     "month": 7,
immich_microservices     |     "day": 19,
immich_microservices     |     "hour": 17,
immich_microservices     |     "minute": 15,
immich_microservices     |     "second": 29,
immich_microservices     |     "tzoffsetMinutes": 0,
immich_microservices     |     "rawValue": "2014:07:19 17:15:29",
immich_microservices     |     "zoneName": "UTC",
immich_microservices     |     "inferredZone": true
immich_microservices     |   },
immich_microservices     |   "TimeScale": 1000,
immich_microservices     |   "PreferredRate": 1,
immich_microservices     |   "PreferredVolume": "100.00%",
immich_microservices     |   "PreviewTime": "0 s",
immich_microservices     |   "PosterTime": "0 s",
immich_microservices     |   "SelectionTime": "0 s",
immich_microservices     |   "CurrentTime": "0 s",
immich_microservices     |   "NextTrackID": 3,
immich_microservices     |   "TrackHeaderVersion": 0,
immich_microservices     |   "TrackCreateDate": "0000:00:00 00:00:00",
immich_microservices     |   "TrackModifyDate": "0000:00:00 00:00:00",
immich_microservices     |   "TrackID": 1,
immich_microservices     |   "TrackLayer": 0,
immich_microservices     |   "TrackVolume": "0.00%",
immich_microservices     |   "ImageWidth": 1280,
immich_microservices     |   "ImageHeight": 720,
immich_microservices     |   "GraphicsMode": "srcCopy",
immich_microservices     |   "OpColor": "0 0 0",
immich_microservices     |   "CompressorID": "avc1",
immich_microservices     |   "SourceImageWidth": 1280,
immich_microservices     |   "SourceImageHeight": 720,
immich_microservices     |   "XResolution": 72,
immich_microservices     |   "YResolution": 72,
immich_microservices     |   "BitDepth": 24,
immich_microservices     |   "VideoFrameRate": 25,
immich_microservices     |   "MatrixStructure": "1 0 0 0 1 0 0 0 1",
immich_microservices     |   "MediaHeaderVersion": 0,
immich_microservices     |   "MediaCreateDate": "0000:00:00 00:00:00",
immich_microservices     |   "MediaModifyDate": "0000:00:00 00:00:00",
immich_microservices     |   "MediaTimeScale": 48000,
immich_microservices     |   "MediaLanguageCode": "und",
immich_microservices     |   "HandlerDescription": "SoundHandler",
immich_microservices     |   "Balance": 0,
immich_microservices     |   "AudioFormat": "mp4a",
immich_microservices     |   "AudioChannels": 2,
immich_microservices     |   "AudioBitsPerSample": 16,
immich_microservices     |   "AudioSampleRate": 48000,
immich_microservices     |   "HandlerType": "Metadata",
immich_microservices     |   "HandlerVendorID": "Apple",
immich_microservices     |   "Encoder": "Lavf53.24.2",
immich_microservices     |   "ImageSize": "1280x720",
immich_microservices     |   "Megapixels": 0.922,
immich_microservices     |   "AvgBitrate": "1.58 Mbps",
immich_microservices     |   "Rotation": 0,
immich_microservices     |   "warnings": []
immich_microservices     | }
immich_microservices     |
immich_microservices     | [Nest] 7  - 03/08/2024, 12:35:52 PM   DEBUG [StorageCore] Attempting to rename file: upload/upload/76a54184-6569-4d47-8c94-1bb24232f6da/6b/9d/6b9d5c77-d8c1-401c-9622-512183b06230.mp4 => upload/library/smartleesh/2024/20240307_105310.mp4
immich_microservices     | [Nest] 7  - 03/08/2024, 12:35:52 PM     LOG [MediaService] Successfully generated JPEG video thumbnail for asset ab56e4e8-cd91-4d81-8311-1393f0ed3c48
immich_machine_learning  | [03/08/24 12:35:52] DEBUG    Available ORT providers: {'AzureExecutionProvider',
immich_machine_learning  |                              'CPUExecutionProvider'}
immich_machine_learning  | [03/08/24 12:35:52] INFO     Setting 'XLM-Roberta-Large-Vit-B-16Plus' execution
immich_machine_learning  |                              providers to ['CPUExecutionProvider'], in
immich_machine_learning  |                              descending order of preference
immich_machine_learning  | [03/08/24 12:35:52] DEBUG    Setting execution provider options to
immich_machine_learning  |                              [{'arena_extend_strategy': 'kSameAsRequested'}]
immich_machine_learning  | [03/08/24 12:35:52] DEBUG    Setting execution_mode to ORT_SEQUENTIAL
immich_machine_learning  | [03/08/24 12:35:52] DEBUG    Setting inter_op_num_threads to 1
immich_machine_learning  | [03/08/24 12:35:52] DEBUG    Setting intra_op_num_threads to 2
immich_machine_learning  | [03/08/24 12:35:52] DEBUG    Setting preferred runtime to onnx
immich_machine_learning  | [03/08/24 12:35:52] INFO     Loading clip model 'XLM-Roberta-Large-Vit-B-16Plus'
immich_machine_learning  |                              to memory
immich_machine_learning  | [03/08/24 12:35:52] DEBUG    Loading clip vision model
immich_machine_learning  |                              'XLM-Roberta-Large-Vit-B-16Plus'
immich_microservices     | [Nest] 7  - 03/08/2024, 12:35:53 PM     LOG [MediaService] Successfully generated WEBP video thumbnail for asset ab56e4e8-cd91-4d81-8311-1393f0ed3c48
immich_machine_learning  | [03/08/24 12:35:53] DEBUG    Loaded clip vision model
immich_machine_learning  |                              'XLM-Roberta-Large-Vit-B-16Plus'
immich_machine_learning  | [03/08/24 12:35:53] DEBUG    Loading tokenizer for CLIP model
immich_machine_learning  |                              'XLM-Roberta-Large-Vit-B-16Plus'
immich_machine_learning  | [03/08/24 12:35:53] DEBUG    Loading model config for CLIP model
immich_machine_learning  |                              'XLM-Roberta-Large-Vit-B-16Plus'
immich_machine_learning  | [03/08/24 12:35:53] DEBUG    Available ORT providers: {'AzureExecutionProvider',
immich_machine_learning  |                              'CPUExecutionProvider'}
immich_machine_learning  | [03/08/24 12:35:53] INFO     Setting 'buffalo_l' execution providers to
immich_machine_learning  |                              ['CPUExecutionProvider'], in descending order of
immich_machine_learning  |                              preference
immich_machine_learning  | [03/08/24 12:35:53] DEBUG    Loaded model config for CLIP model
immich_machine_learning  |                              'XLM-Roberta-Large-Vit-B-16Plus'
immich_machine_learning  | [03/08/24 12:35:53] DEBUG    Loading tokenizer config for CLIP model
immich_machine_learning  |                              'XLM-Roberta-Large-Vit-B-16Plus'
immich_machine_learning  | [03/08/24 12:35:53] DEBUG    Setting execution provider options to
immich_machine_learning  |                              [{'arena_extend_strategy': 'kSameAsRequested'}]
immich_machine_learning  | [03/08/24 12:35:53] DEBUG    Setting execution_mode to ORT_SEQUENTIAL
immich_machine_learning  | [03/08/24 12:35:53] DEBUG    Loaded tokenizer config for CLIP model
immich_machine_learning  |                              'XLM-Roberta-Large-Vit-B-16Plus'
immich_machine_learning  | [03/08/24 12:35:53] DEBUG    Setting inter_op_num_threads to 1
immich_machine_learning  | [03/08/24 12:35:54] DEBUG    Setting intra_op_num_threads to 2
immich_machine_learning  | [03/08/24 12:35:54] DEBUG    Setting preferred runtime to onnx
immich_machine_learning  | [03/08/24 12:35:54] DEBUG    Loaded tokenizer for CLIP model
immich_machine_learning  |                              'XLM-Roberta-Large-Vit-B-16Plus'
immich_machine_learning  | [03/08/24 12:35:54] DEBUG    Loading visual preprocessing config for CLIP model
immich_machine_learning  |                              'XLM-Roberta-Large-Vit-B-16Plus'
immich_machine_learning  | [03/08/24 12:35:54] DEBUG    Loaded visual preprocessing config for CLIP model
immich_machine_learning  |                              'XLM-Roberta-Large-Vit-B-16Plus'
immich_machine_learning  | [03/08/24 12:35:54] INFO     Loading facial recognition model 'buffalo_l' to
immich_machine_learning  |                              memory
immich_microservices     | [Nest] 7  - 03/08/2024, 12:35:55 PM   DEBUG [PersonService] 0 faces detected in upload/thumbs/76a54184-6569-4d47-8c94-1bb24232f6da/ab/56/ab56e4e8-cd91-4d81-8311-1393f0ed3c48.jpeg
immich_microservices     | [Nest] 7  - 03/08/2024, 12:35:55 PM VERBOSE [PersonService] Object:
immich_microservices     | []
immich_microservices     |

### Logs generated when a photo was uploaded
immich_microservices     | [Nest] 7  - 03/08/2024, 12:39:47 PM VERBOSE [MetadataService] Exif Tags
immich_microservices     | [Nest] 7  - 03/08/2024, 12:39:47 PM VERBOSE [MetadataService] Object:
immich_microservices     | {
immich_microservices     |   "SourceFile": "/usr/src/app/upload/upload/76a54184-6569-4d47-8c94-1bb24232f6da/5b/72/5b724f83-3341-4a47-bb24-2acba3be1c77.png",
immich_microservices     |   "errors": [],
immich_microservices     |   "ExifToolVersion": 12.76,
immich_microservices     |   "FileName": "5b724f83-3341-4a47-bb24-2acba3be1c77.png",
immich_microservices     |   "Directory": "/usr/src/app/upload/upload/76a54184-6569-4d47-8c94-1bb24232f6da/5b/72",
immich_microservices     |   "FileSize": "577 kB",
immich_microservices     |   "FileModifyDate": {
immich_microservices     |     "_ctor": "ExifDateTime",
immich_microservices     |     "year": 2024,
immich_microservices     |     "month": 3,
immich_microservices     |     "day": 7,
immich_microservices     |     "hour": 20,
immich_microservices     |     "minute": 15,
immich_microservices     |     "second": 25,
immich_microservices     |     "tzoffsetMinutes": 540,
immich_microservices     |     "rawValue": "2024:03:07 20:15:25+09:00",
immich_microservices     |     "zoneName": "UTC+9",
immich_microservices     |     "inferredZone": false
immich_microservices     |   },
immich_microservices     |   "FileAccessDate": {
immich_microservices     |     "_ctor": "ExifDateTime",
immich_microservices     |     "year": 2024,
immich_microservices     |     "month": 3,
immich_microservices     |     "day": 8,
immich_microservices     |     "hour": 12,
immich_microservices     |     "minute": 39,
immich_microservices     |     "second": 47,
immich_microservices     |     "tzoffsetMinutes": 540,
immich_microservices     |     "rawValue": "2024:03:08 12:39:47+09:00",
immich_microservices     |     "zoneName": "UTC+9",
immich_microservices     |     "inferredZone": false
immich_microservices     |   },
immich_microservices     |   "FileInodeChangeDate": {
immich_microservices     |     "_ctor": "ExifDateTime",
immich_microservices     |     "year": 2024,
immich_microservices     |     "month": 3,
immich_microservices     |     "day": 8,
immich_microservices     |     "hour": 12,
immich_microservices     |     "minute": 39,
immich_microservices     |     "second": 47,
immich_microservices     |     "tzoffsetMinutes": 540,
immich_microservices     |     "rawValue": "2024:03:08 12:39:47+09:00",
immich_microservices     |     "zoneName": "UTC+9",
immich_microservices     |     "inferredZone": false
immich_microservices     |   },
immich_microservices     |   "FilePermissions": "-rw-r--r--",
immich_microservices     |   "FileType": "PNG",
immich_microservices     |   "FileTypeExtension": "png",
immich_microservices     |   "MIMEType": "image/png",
immich_microservices     |   "ImageWidth": 1362,
immich_microservices     |   "ImageHeight": 797,
immich_microservices     |   "BitDepth": 8,
immich_microservices     |   "ColorType": "RGB with Alpha",
immich_microservices     |   "Compression": "Deflate/Inflate",
immich_microservices     |   "Filter": "Adaptive",
immich_microservices     |   "Interlace": "Noninterlaced",
immich_microservices     |   "SRGBRendering": "Perceptual",
immich_microservices     |   "Gamma": 2.2,
immich_microservices     |   "PixelsPerUnitX": 4724,
immich_microservices     |   "PixelsPerUnitY": 4724,
immich_microservices     |   "PixelUnits": "meters",
immich_microservices     |   "SnipMetadata": "{\"clipPoints\":[{\"x\":0,\"y\":0},{\"x\":1363,\"y\":0},{\"x\":1363,\"y\":797},{\"x\":0,\"y\":797}]}",
immich_microservices     |   "ImageSize": "1362x797",
immich_microservices     |   "Megapixels": 1.1,
immich_microservices     |   "warnings": []
immich_microservices     | }
immich_microservices     |
immich_microservices     | [Nest] 7  - 03/08/2024, 12:39:47 PM   DEBUG [StorageCore] Attempting to rename file: upload/upload/76a54184-6569-4d47-8c94-1bb24232f6da/5b/72/5b724f83-3341-4a47-bb24-2acba3be1c77.png => upload/library/smartleesh/2024/20240307_201525.png
immich_microservices     | [Nest] 7  - 03/08/2024, 12:39:47 PM     LOG [MediaService] Successfully generated JPEG image thumbnail for asset 4f6b0924-7822-4434-a311-dc286cb67240
immich_microservices     | [Nest] 7  - 03/08/2024, 12:39:47 PM     LOG [MediaService] Successfully generated WEBP image thumbnail for asset 4f6b0924-7822-4434-a311-dc286cb67240
immich_microservices     | [Nest] 7  - 03/08/2024, 12:39:48 PM   DEBUG [PersonService] 1 faces detected in upload/thumbs/76a54184-6569-4d47-8c94-1bb24232f6da/4f/6b/4f6b0924-7822-4434-a311-dc286cb67240.jpeg
immich_microservices     | [Nest] 7  - 03/08/2024, 12:39:48 PM VERBOSE [PersonService] Object:
immich_microservices     | [
immich_microservices     |   {
immich_microservices     |     "imageWidth": 1362,
immich_microservices     |     "imageHeight": 797,
immich_microservices     |     "boundingBox": {
immich_microservices     |       "x1": 419,
immich_microservices     |       "y1": 105,
immich_microservices     |       "x2": 495,
immich_microservices     |       "y2": 195
immich_microservices     |     },
immich_microservices     |     "score": 0.8274691104888916,
immich_microservices     |     "embedding": "vector(512)"
immich_microservices     |   }
immich_microservices     | ]
immich_microservices     |
immich_microservices     | [Nest] 7  - 03/08/2024, 12:39:48 PM VERBOSE [JobRepository] Waiting for undefined queue to stop...
immich_machine_learning  | [03/08/24 12:39:48] DEBUG    Checking for inactivity...
immich_microservices     | [Nest] 7  - 03/08/2024, 12:39:49 PM   DEBUG [PersonService] Skipping facial recognition queueing because 1 job is already queued
immich_microservices     | [Nest] 7  - 03/08/2024, 12:39:49 PM   DEBUG [PersonService] Face 3f29909f-d4dc-4b78-b33d-0befac60fa9b only matched the face itself, skipping

**Comment by jenda69** [comment]
I noticed that video metadata handling (esp. timezones) is much weirder for videos too, even if they are taken with the same camera as surrounding photos.
However I found that there's a workaround that if there's a sidecar for the video file, the timezone shows up correctly in Immich.

I historically use an external library with a script to handle the directory structure, files naming etc. so I just added `exiftool -ext mp4 -o %d%f.%e.xmp` to the script to generate the sidecars for videos and that solved the issue for me.

I remember reading it might be because some QuickTime metadata are expected to be always in UTC but many cameras don't really adhere to that.

**Comment by DeltaTango69** [comment]
Hi - I refer to this thread

Time offset in timeline UTC - local time #6799

For me the solution was to eliminate the wrong xmp metadata. Exiftools seems to deliver first the xmp createdate. After deleting this entry the quicktime createdate is used and this entry should be only in UTC Time.

**Comment by mstewart14** [comment]
me too... that is to say having the same problem with videos from the phone end up marked as UTC despite correct timezone appearing on photos from same shoot.
